client/adventureMap/TurnTimerWidget.cpp

```diff
@@ -95,10 +98,22 @@ void TurnTimerWidget::tick(uint32_t msPassed)
 	if(LOCPLINT && LOCPLINT->cb)
 	{
 		auto player = LOCPLINT->cb->getCurrentPlayer();
+		if(LOCPLINT->battleInt)
+		{
+			if(auto * stack = LOCPLINT->battleInt->stacksController->getActiveStack())
+				player = stack->getOwner();
```

Comment:
Does this method returns correct owner for hypnotized units?

Reply:
CStack has owner field, getOwner is getter for this field